container-storage-setup.sh

```diff
@@ -1002,13 +1046,9 @@ get_current_storage_options() {
 is_valid_storage_driver() {
   local driver=$1 d
 
-  # Empty driver is valid. That means user does not want us to setup any
-  # storage.
-  [ -z "$driver" ] && return 0
-
-  for d in $_STORAGE_DRIVERS;do
-    [ "$driver" == "$d" ] && return 0
-  done
+  case "$driver" in
+      ""|auto|devicemapper|overlay|overlay2) return 0;;
```

Comment:
I reallly don't understand this. defining all valid storage drivers in a single global variable makes so much of sense. Rest of the code can make use of it.

Why to move it inside a function. What do we gain.

Reply:
Before, the only user of that variable was `is_valid_storage_driver` - every *other* function needs to take action depending on the type by using `case` - so we're now more consistent.